Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp

```diff
@@ -1240,7 +1240,7 @@ class CombinatorialKalmanFilter {
         ACTS_VERBOSE(
             "Reverse navigation direction after smoothing for reaching the "
             "target surface");
-        state.stepping.navDir = state.stepping.navDir.invert();
+        state.options.direction = state.options.direction.invert();
```

Comment:
Oh ok, so we do reverse indeed. Seems a bit odd that the input options are not immutable. Otoh they do get copied in, so the state does de-facto own them.

Reply:
it also bugs me a bit that it changes the "options" now but on the other hand this is how the propagation currently works. as you point out it is a copy so ownership should be fine